tdct/deconvolution.py

```diff
@@ -0,0 +1,864 @@
+        vmax= d0.max()
+        vmin=d0.min()
+        d2 = (d1-vmin)/(vmax-vmin)
+        ret=d2
+    return ret
+
+def _centered(arr, newshape):
+    # Return the center newshape portion of the array.
+    newshape = np.asarray(newshape)
+    currshape = np.array(arr.shape)
+    startind = (currshape - newshape) // 2
+    endind = startind + newshape
+    myslice = [slice(startind[k], endind[k]) for k in range(len(endind))]
+    return arr[tuple(myslice)]
+
+def doRLDeconvolution(datapath , psfdatapath , niter=0, qtprocessbar=None):
+    #Internal class to handle progress bar
+
+    ''' This code is not giving correct results
+    This is probably because the usage of the psf flipped
+    In a later version doRLDeconvolution5 (based on DeconvolutionLab2 instead of using psf-flipped in the 2nd convolution,
+    a correlation is performed. '''
+
+    #Estimate progress iterations
+    nProgrIter = 2*niter + 6 #Check if ok
+    progr0 = _progrBarHandle(qtprocessbar, nProgrIter) #Sets up
+
+    #Read data
+    if os.path.isfile(datapath) is True and os.path.isfile(psfdatapath) is True and niter>0:
+        if debug is True: print(clrmsg.DEBUG, "Loading images: ", datapath," , ",psfdatapath)
+        progr0.increment() #1
+
+        data_np = tf.imread(datapath)
+        psf_np = tf.imread(psfdatapath)
+
+        progr0.increment() #2
+
+        #Normalise both data (and also converts)
+
+        data_np_norm = _convertAndNormalise(data_np)
+        psf_np_norm = _convertAndNormalise(psf_np)
+
+        progr0.increment() #3
+
+        #Code inspired by scikit-image Richardson-lucy
+        #https://github.com/scikit-image/scikit-image/blob/main/skimage/restoration/deconvolution.py#L383-L443
+        psf_t = np.flip(psf_np_norm)
+        xn1 = np.full(data_np_norm.shape,0.5, dtype=data_np_norm.dtype)
+
+        progr0.increment() #4
+        
+        if debug is True: print(clrmsg.DEBUG, "Begining iterative deconvolution niter=", niter)
+
+        for i in range(niter):
+            xn=xn1
+            
+            Hx = scipy.signal.convolve(xn,psf_np_norm, mode='same')
+            yhx = np.divide(data_np_norm,Hx)
+
+            progr0.increment()
+
+            htyhx = scipy.signal.convolve(yhx,psf_t, mode='same')
+            xn1 = np.multiply(xn, htyhx)
+
+            progr0.increment()
+
+        #At the end xn1 should have the deconvoluted data
+        #Convert to uint8
+        data_deconv_norm_256 = _convertAndNormalise(xn1)*256
+        data_deconv_uint8 = data_deconv_norm_256.astype('uint8')
+        
+        progr0.increment() #5
+
+        #Save data
+        fpath,fname = os.path.split(datapath)
+        fname0 = os.path.join(fpath, "DeconvRL"+str(niter)+"it_"+fname)
+        tf.imsave(fname0 ,data_deconv_uint8)
+
+        progr0.setmax()
+ 
+        if debug is True: print(clrmsg.DEBUG, "Completed deconvolution, file saved: ", fname0)
+
+def doRLDeconvolution2(datapath , psfdatapath , niter=0, qtprocessbar=None):
```

Comment:
Again naming and doc strings could be better. Also, I guess this is a work in progress? If doRLDeconvolution2 is better than doRLDeconvolution then maybe delete the first one.